addon/components/paper-button.js

```diff
@@ -4,45 +4,32 @@ import RippleMixin from '../mixins/ripple-mixin';
 import ProxiableMixin from 'ember-paper/mixins/proxiable-mixin';
 import ColorMixin from 'ember-paper/mixins/color-mixin';
 
+const {
+  assert,
+  computed,
+  isPresent
+} = Ember;
+
 export default BaseFocusable.extend(RippleMixin, ProxiableMixin, ColorMixin, {
-  attributeBindings: ['target', 'action', 'type'],
+  attributeBindings: ['type'],
+  type: 'button',
   tagName: 'button',
-  themed: true,
-  classNameBindings: ['raised:md-raised', 'icon-button:md-icon-button', 'focus:md-focused', 'themed:md-default-theme', 'themed:md-button'],
+  classNames: ['paper-button', 'md-default-theme', 'md-button'],
+  classNameBindings: [
+    'raised:md-raised',
+    'iconButton:md-icon-button',
+  ],
 
   // Ripple Overrides
   rippleContainerSelector: null,
-  fitRipple: Ember.computed.readOnly('isIconButton'),
-  center: Ember.computed.readOnly('isIconButton'),
-  dimBackground: Ember.computed.not('isIconButton'),
-
-  noSpan: Ember.computed('no-span', function() {
-    return this.get('no-span');
-  }),
-
-  // RippleMixin overrides
-  isIconButton: Ember.computed(function() {
-    return this.classNames.any(function(className) {
-      return className.indexOf('md-icon-button') !== -1;
-    });
-  }),
-  isMenuItem: Ember.computed(function() {
-    return this.classNames.any(function(className) {
-      return className.indexOf('md-menu-item') !== -1;
-    });
-  }),
+  fitRipple: computed.readOnly('iconButton'),
+  center: computed.readOnly('iconButton'),
+  dimBackground: computed.not('iconButton'),
 
-  // bubble actions by default
-  bubbles: true,
   click(event) {
-    let target = this.get('target');
-
-    if (target) {
-      this.get('target').send(this.get('action'), this.get('param'), event);
-    } else {
-      this.sendAction('action', this.get('param'), event);
+    let onClick = this.get('onClick');
+    if (isPresent(onClick)) {
+      this.get('onClick')(event);
```

Comment:
Please use `this.sendAction('onClick')`:
1. it is compatible with closure actions
2. you don't need to check if an action exists that way